### PR TITLE
Astro Refactoring 

### PR DESCRIPTION
**Pull request by HankG**
An example refactoring of the Astro code from BB into this project to show which parts were ported.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -18,6 +18,22 @@
...
   <dependencies>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <!--       Explicit dependency on this version needed to avoid NoSuchMethod exceptions. -->
+      <version>20.0</version>
+    </dependency>
+    <dependency>
+      <groupId>org.b612foundation.adam</groupId>
+      <artifactId>StkParts</artifactId>
```
Comment (HoodedCrow) [inline pom.xml]: Let's try to avoid pulling this in even by mention.

File: pom.xml [hunk 2]
```diff
@@ -18,6 +18,22 @@
   </properties>
 
   <dependencies>
+    <dependency>
+      <groupId>com.google.guava</groupId>
+      <artifactId>guava</artifactId>
+      <!--       Explicit dependency on this version needed to avoid NoSuchMethod exceptions. -->
```
Comment (HoodedCrow) [inline pom.xml]: Where is the exception coming from?

File: src/main/java/org/b612foundation/adam/datamodel/AdamRunnableException.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package org.b612foundation.adam.runnable;
+
+/**
+ * For use when errors arise while generating/performing calculations for AdamObjects.
+ * 
+ * Note: This should live somewhere else, possibly in astrodynamics.
+ */
+public class AdamRunnableException extends Exception {
```
Comment (HoodedCrow) [inline src/main/java/org/b612foundation/adam/datamodel/AdamRunnableException.java]: As discussed elsewhere, this should not be here.

File: src/main/java/org/b612foundation/adam/propagators/ClasspathStreamFactory.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+import java.io.IOException;
+import java.io.InputStream;
+
+import org.apache.commons.io.IOUtils;
+
+import agi.foundation.compatibility.MemoryStream;
+import agi.foundation.infrastructure.StreamFactory;
+
+public class ClasspathStreamFactory extends StreamFactory {
```
Comment (HoodedCrow) [inline src/main/java/org/b612foundation/adam/propagators/ClasspathStreamFactory.java]: May not be needed here at all, or at least should get bundled with other STK stuff, possibly into a subpackage.

**Comment by HankG** [comment]
This is OBE since this was just an experiment and we have what we need from it. Deleting it and the branch.

**Comment by HankG** [comment]
This is OBE since this was just an experiment and we have what we need from it. Deleting it and the branch.

**Pull request closed without merging**